src/compas/files/base_reader.py

```diff
@@ -0,0 +1,202 @@
+
+        Returns
+        -------
+        bool
+            True if binary else false.
+        """
+        return binaryornot.check.is_binary(str(self.location))
+
+    @property
+    def is_valid(self):
+        return NotImplementedError
+
+    def is_adress_url(self):
+        """Checks if given location is a string containing an URL
+
+        Returns
+        -------
+        bool
+            True if recognized as an URL
+        """
+        return str(self._adress).startswith('http')
+
+    def _download(self, url):
+        """Downloads file and returns path to tempfle
+
+        Called by property self.location
+
+        Parameters
+        ----------
+        url : string
+            URL to file
+
+        Returns
+        -------
+        location : Pathlib object
+            Path to tempfile
+        """
+        location, _ = urlretrieve(url)
+
+        return Path(location)
+
+    def open_ascii(self):
+        """Open ascii file and return file object
+
+        Returns
+        -------
+        file object
+        """
+        try:
+            file_object = self.location.open(mode='r')
+        except UnicodeDecodeError:
+            file_object = self.location.open(mode='r', errors='replace', newline='\r')
+        return file_object
+
+    def open_binary(self):
+        """Open binary file and return file object
+
+        Returns
+        -------
+        file object
+        """
+        return self.location.open(mode='rb')
+
+    def iter_lines(self):
+        """Yields lines from ascii file
+
+        Yields
+        -------
+        string
+            Next line or chunk of file
+        """
+        # TODO: Handle continuing lines (as in OFF files)
+
+        with self.open_ascii() as fo:
+            for line in fo:
+                yield line.rstrip()
+
+    def iter_chunks(self, chunk_size=4096):
+        """Yields lines from binary files
+
+        Parameters
+        ----------
+        chunk_size : int
+            Chunks to read with each call
+
+        Yields
+        ------
+        bytes
+            Next chunk of file
+        """
+        with self.open_binary() as fo:
+            for chunk in iter(lambda: fo.read(chunk_size), b''):
+                yield chunk
+
+    def read(self):
+        raise NotImplementedError
+
+    def is_file_signature_correct(self):
+        """Checks wether file signature (also known as magic number) is present
+        in input file.
+
+        File signatures are strings, numbers or bytes defined in a file
+        format's specification. While not technically required to parse the
+        file, a missing file signatures might be a sign of a malformed file.
+
+        More information about file signatures can be found on Wikipedia[1]_
+        as well as examples of file signatures[2]_
+
+        Returns
+        ------
+        bool
+            True file signature for file type is found in file or if file type
+            lacks file signature.
```

Comment:
True if correct file signature for file type is found in file, or if file type has no signature.